### PR TITLE
closes #195

### LINKED ISSUE #195: salsa20 polymorphic
### Inputs and outputs

This is our first
[polymorphic](https://en.wikipedia.org/wiki/Polymorphism_(computer_science))
function. In case you need a refresher, a polymorphic function is one
that can take more than one type of input. In this case, we have to
take two different types of `k` --- one that is a 32-byte sequence and
one that is a 16-byte sequence. Here we create a _type variable_ `a`
such that when `a` is one, the function expects a 16-byte `k` and when
it's two, the function expects a 32-byte `k`. We also constrain `k`
(using _type constraints_) to only be able to take on the value one or
two.

```cryptol
Salsa20Expansion :
    {a}
    (a >= 1, 2 >= a) =>
    Bytes (16*a) -> Bytes 16 -> Bytes 64
```

***********
It states that this is the first polymorphic function.  however earlier in the module littleendian' is specifically made to be polymorphic.

### PR DESCRIPTION
**Pull request by weaversa**

### CODE REVIEW

File: labs/Salsa20/Salsa20Answers.md [hunk 1]
```diff
@@ -696,14 +696,14 @@ Nothing of consequence
...
-that can take more than one type of input. In this case, we have to
-take two different types of `k` --- one that is a 32-byte sequence and
-one that is a 16-byte sequence. Here we create a _type variable_ `a`
-such that when `a` is one, the function expects a 16-byte `k` and when
-it's two, the function expects a 32-byte `k`. We also constrain `k`
-(using _type constraints_) to only be able to take on the value one or
-two.
+function (well, the first one of consequence). A polymorphic function
+is one that can take more than one type of input. In this case, we
+have to take two different types of `k` --- one that is a 32-byte
```
Comment (WeeknightMVP) [inline labs/Salsa20/Salsa20Answers.md:701]: The `---`s here and elsewhere are also throwing me, but that's a whole 'nother issue.

File: labs/Salsa20/Salsa20Answers.md [hunk 2]
```diff
@@ -696,14 +696,14 @@ Nothing of consequence
...
+sequence and one that is a 16-byte sequence. Here we create a _type
```
Comment (WeeknightMVP) [inline labs/Salsa20/Salsa20Answers.md:702]: I think `a` is `1` or `2`, not "one"....

File: labs/Salsa20/Salsa20Answers.md [hunk 3]
```diff
@@ -696,14 +696,14 @@ Nothing of consequence
...
+variable_ `a` such that when `a` is one, the function expects a
+16-byte `k` and when it's two, the function expects a 32-byte `k`. We
```
Comment (WeeknightMVP) [inline labs/Salsa20/Salsa20Answers.md:704]: ...or "two"...

File: labs/Salsa20/Salsa20Answers.md [hunk 4]
```diff
@@ -696,14 +696,14 @@ Nothing of consequence
...
+also constrain `k` (using _type constraints_) to only be able to take
+on the value one or two.
```
Comment (WeeknightMVP) [inline labs/Salsa20/Salsa20Answers.md:706]: .

**Review by WeeknightMVP — APPROVED** [review]

File: labs/Salsa20/Salsa20Answers.md (hunk 1 above)
Comment (WeeknightMVP) [reply labs/Salsa20/Salsa20Answers.md:701]: I found a reliable source that `-` is a hyphen, `--` is an _en dash_, and `---` represents an _em dash_.

Comment (weaversa) [reply labs/Salsa20/Salsa20Answers.md:701]: eh...take it or leave it.

**Pull request merged**